src/shared/auth.ts

```diff
@@ -16,6 +18,13 @@ export const authOptions: NextAuthOptions = {
   pages: {
     signIn: "/login",
   },
+  callbacks: {
+    async signIn(user) {
+      const loginUser = user.user as AdapterUser;
+      createUser(loginUser);
+      return true;
```

Comment:
looks like you registering/creating the user, that means that the func should be called singUp instead of singIn. [src](https://fisk.com.br/blog/qual-e-a-diferenca-entre-sign-up-sign-in-e-sign-out)

Reply:
This is a standard callback function called whenever the user signs in. I didn't create it, it's just one of the options offered by NextAuth as a callback.

See docs: https://next-auth.js.org/configuration/callbacks